### PR TITLE
Add query test and docs

### PR DESCRIPTION
**Pull request by ssh24**
### Description

Add tests cases to query different operations in dashdb.

connect to https://github.com/strongloop/loopback-connector-dashdb/issues/51

**Please note:** The tests will fail. Needs fix https://github.com/strongloop/loopback-ibmdb/pull/57 to land first.

### CODE REVIEW

File: test/dashdb.query.test.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+          function(err, posts) {
+            assert(!err);
+            assert(posts);
+            assert.equal(posts.length, 1);
+            assert.equal(posts[0].content, data[0].content);
+            done();
+          });
+      });
+
+      it('as regex literal', function(done) {
```
Comment (b-admike) [inline test/dashdb.query.test.js]: Duplicate of L66-L75 tests? If so, you can remove.

Comment (ssh24) [inline test/dashdb.query.test.js]: Not really. Those tests are using `using no flags` and this set is `using flags`.

Comment (b-admike) [inline test/dashdb.query.test.js]: Ah, right. LGTM.

**Comment by ssh24** [comment]
@slnode test please

**Pull request merged**